specify/specify7
# Implement prefs and other functionalities for Tectonic trees

Fixes #5333

- [x] Add a count stat to Tectonic trees to make `TreeEditor.Rank.Threshold.TectonicUnit` pref work (info: https://github.com/specify/specify7/pull/5318#discussion_r1799887113)
- [x] Verify `queryFromTree` works and fix unit tests for those (info: https://github.com/specify/specify7/pull/5318#discussion_r1797332831)
- [x] Add static data for Tectonic trees and verify tree unit tests work

- Added workbench scoping for Tectonic trees
- Added tree hierarchy context for Tectonic trees

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions



- **Use a paleo db**
- Go to Trees > Tectonic Unit
  - [ ] Verify a node can be added
  - [ ] Verify a node can be removed
  - [ ] Verify a rank can be added
  - [ ] Verify a rank can be removed
  - [ ] Tree can be renamed
  - [ ] Verify a node can be searched
  - Test other tree functionalities if any
 - Go to Query Builder
   - [ ] Verify queries work for Tectonic Unit 
   - [ ] Verify queries work for Tectonic Unit Tree Definition
   - [ ] Verify queries work for Tectonic Unit Tree Definition Items
- Go to workbench
  - [ ] Verify Tectonic Unit can be added via Workbench
